### PR TITLE
Tw 1162/paste multiple images from os

### PR DESCRIPTION
**Pull request by sherlockvn**
### Demo:

https://github.com/linagora/twake-on-matrix/assets/43041967/9bf12e59-4353-4982-978d-645847af298e

### CODE REVIEW

File: lib/pages/chat/input_bar/input_bar.dart [hunk 1]
```diff
@@ -334,109 +335,63 @@ class InputBar extends StatelessWidget with PasteImageMixin {
...
       room: room,
       onEnter: _onEnter,
-      handlePaste: enablePasteImage ? () => handlePaste(context) : null,
-      child: ContextMenuInputBar(
+      child: TypeAheadField<Map<String, String?>>(
+        direction: VerticalDirection.up,
+        hideOnEmpty: true,
+        hideOnLoading: true,
+        hideOnSelect: false,
+        debounceDuration: const Duration(milliseconds: 50),
```
Comment (nqhhdev) [inline lib/pages/chat/input_bar/input_bar.dart:343]: Create const

File: lib/pages/chat/input_bar/input_bar.dart [hunk 2]
```diff
@@ -334,109 +335,63 @@ class InputBar extends StatelessWidget with PasteImageMixin {
...
+          controller: controller,
+          decoration: decoration,
+          focusNode: focusNode,
+          onChanged: (text) {
+            if (onChanged != null) {
+              onChanged!(text);
+            }
+          },
+          onTap: () async {
+            await Future.delayed(const Duration(milliseconds: 100));
```
Comment (nqhhdev) [inline lib/pages/chat/input_bar/input_bar.dart:364]: Create const

**Comment by nqhhdev** [comment]
Pls fetch and rebase on `upgrade_3.16` branch

File: lib/presentation/mixins/paste_image_mixin.dart [hunk 3]
```diff
@@ -1,55 +1,46 @@
...
       TwakeSnackBar.show(context, L10n.of(context)!.pasteImageFailed);
       return;
     }
-
+    final nonNullableFiles = matrixFiles
+        .where(
+          (element) => element != null,
+        )
+        .cast<MatrixFile>()
+        .toList();
```
Comment (nqhhdev) [inline lib/presentation/mixins/paste_image_mixin.dart:36]:
```suggestion
    final nonNullableFiles = matrixFiles
        .where(
          (matrixFile) => matrixFile != null,
        )
        .cast<MatrixFile>()
        .toList();
```

File: lib/pages/chat/input_bar/input_bar.dart (hunk 1 above)
Comment (sherlockvn) [reply lib/pages/chat/input_bar/input_bar.dart]: done

File: lib/pages/chat/input_bar/input_bar.dart (hunk 2 above)
Comment (sherlockvn) [reply lib/pages/chat/input_bar/input_bar.dart]: done

File: lib/presentation/mixins/paste_image_mixin.dart (hunk 3 above)
Comment (sherlockvn) [reply lib/presentation/mixins/paste_image_mixin.dart]: done

**Review by hoangdat — APPROVED** [review]

**Review by nqhhdev — APPROVED** [review]

**Comment by sherlockvn** [comment]
Wait. Let me check it works well in Firefox

**Comment by sherlockvn** [comment]
> Wait. Let me check it works well in Firefox
copy multiple images in firefox is not work yet. 
Create a ticket for later: https://github.com/linagora/twake-on-matrix/issues/1263

**Pull request merged**